crates/bevy_pbr/src/wireframe.rs

```diff
@@ -46,7 +46,9 @@ impl Plugin for WireframePlugin {
                     global_color_changed.run_if(resource_changed::<WireframeConfig>),
                     wireframe_color_changed,
                     apply_wireframe_material,
-                    apply_global_wireframe_material.run_if(resource_changed::<WireframeConfig>),
+                    // Run after `apply_wireframe_material` so that the global wireframe setting is applied
+                    // to a mesh on the same frame its wireframe marker component is removed.
+                    apply_global_wireframe_material.after(apply_wireframe_material),
```

Comment:
I would suggest to use `chain`: `(apply_wireframe_material, apply_global_wireframe_material).chain()`